R/glmClassical.R

```diff
@@ -0,0 +1,1223 @@
+    xLimits <- range(xBreaks)
+    thePlot <- ggplot2::ggplot(mapping = ggplot2::aes(x = x, y = y),
+                               data = d) +
+      jaspGraphs::geom_point() +
+      ggplot2::geom_smooth(se = FALSE,
+                           size = 0.6,
+                           method = "loess",
+                           method.args = list(degree = 1, family = "symmetric")) +
+      ggplot2::xlab(gettext(predictor)) +
+      ggplot2::ylab(gettextf("Standardized %1s residual", residType)) +
+      ggplot2::scale_x_continuous(breaks = xBreaks, limits = xLimits) +
+      ggplot2::scale_y_continuous(breaks = yBreaks, limits = yLimits) +
+      jaspGraphs::geom_rangeframe() +
+      jaspGraphs::themeJaspRaw()
+  }
+  return(thePlot)
+}
+
+
+
+# Plots: Residuals Q-Q
+.glmPlotResQQ <- function(jaspResults, dataset, options, ready, position) {
+
+  plotNames <- c("devResQqPlot", "prsResQqPlot", "quanResQqPlot")
+  if (!ready || !any(unlist(options[plotNames])))
+    return()
+
+  residNames <- c("deviance", "Pearson", "quantile")
+
+  glmPlotResQQContainer <- createJaspContainer(gettext("Normal Q-Q Plots: Standardized Residuals"))
+  glmPlotResQQContainer$dependOn(optionsFromObject = jaspResults[["modelSummary"]],
+                                 options           = plotNames)
+  glmPlotResQQContainer$position <- position
+  jaspResults[["diagnosticsContainer"]][["glmPlotResQQ"]] <- glmPlotResQQContainer
+
+
+  if (!is.null(jaspResults[["glmModels"]])) {
+    glmFullModel <- jaspResults[["glmModels"]][["object"]][["fullModel"]]
+    for (i in 1:length(plotNames)) {
+      if (options[[plotNames[[i]]]]) {
+        .glmCreatePlotPlaceholder(glmPlotResQQContainer,
+                                  index = plotNames[[i]],
+                                  title = gettextf("Normal Q-Q plot: Standardized %1s residuals", residNames[[i]]))
+
+        .glmInsertPlot(glmPlotResQQContainer[[plotNames[[i]]]],
+                       .glmFillPlotResQQ,
+                       residType = residNames[[i]],
+                       model = glmFullModel,
+                       family = options[["family"]])
+      }
+    }
+  }
+  return()
+}
+
+.glmFillPlotResQQ <- function(residType, model, family) {
+
+  # compute residuals
+  stdResid <- .glmStdResidCompute(model = model, residType = residType)
+
+  # make plot
+  breaks <- pretty(stdResid)
+  limits <- range(breaks) #note that the breaks and limits work for both x and y axis
+  thePlot <- ggplot2::ggplot(mapping = ggplot2::aes(sample = y),
+                             data = data.frame(y = stdResid)) +
+    ggplot2::stat_qq() +
+    ggplot2::xlab(gettext("Theoretical Quantiles")) +
+    ggplot2::ylab(gettext("Sample Quantiles")) +
+    ggplot2::scale_x_continuous(breaks = breaks) +
+    ggplot2::scale_y_continuous(breaks = breaks) +
+    ggplot2::coord_cartesian(xlim = limits, ylim = limits) +
+    ggplot2::stat_qq_line() +
+    jaspGraphs::geom_rangeframe() +
+    jaspGraphs::themeJaspRaw()
```

Comment:
```suggestion
   plot <- jaspGraphs::plotQQnorm(stdResid)
```
Seems like you could simplify the code by reusing `jaspGraphs` function... And this also ensures that the styling is consistent

Reply:
Ah ok, will try that!